app/Http/Controllers/CardDAV/MonicaSabreBackend.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace App\Http\Controllers\CardDAV;
+
+use Sabre\DAV, Sabre\CalDAV, Sabre\DAVACL;
+use Auth;
+use Log;
+
+class MonicaSabreBackend implements \Sabre\DAV\Auth\Backend\BackendInterface {
+
+    /**
+     * When this method is called, the backend must check if authentication was
+     * successful.
+     *
+     * The returned value must be one of the following
+     *‹
+     * [true, "principals/username"]
+     * [false, "reason for failure"]
+     *
+     * If authentication was successful, it's expected that the authentication
+     * backend returns a so-called principal url.
+     *
+     * Examples of a principal url:
+     *
+     * principals/admin
+     * principals/user1
+     * principals/users/joe
+     * principals/uid/‹123457
+     *
+     * If you don't use WebDAV ACL (RFC3744) we recommend that you simply
+     * return a string such as:
+     *
+     * principals/users/[username]
+     *
+     * @param RequestInterface $request
+     * @param ResponseInterface $response
+     * @return array
+     */
+    function check(\Sabre\HTTP\RequestInterface $request, \Sabre\HTTP\ResponseInterface $response) {
+        Log::debug(__CLASS__.' check', func_get_args());
+        if (Auth::check() || Auth::attempt(['email' => $request->getRawServerValue('PHP_AUTH_USER'), 'password' => $request->getRawServerValue('PHP_AUTH_PW')])) {
```

Comment:
Might be easier to put `Auth::attempt(['email' => $request->getRawServerValue('PHP_AUTH_USER'), 'password' => $request->getRawServerValue('PHP_AUTH_PW')])` in a variable so this line is more readable.

Reply:
I agree. I might revert back to using https://laravel.com/docs/5.6/authentication#stateless-http-basic-authentication I prefer using the framework systems whenever possible, but I need to figure out the issue with sync first.